### ISSUE TITLE
Improve User retrieval.

### ISSUE THREAD

**Issue by richardkmichael** [richardkmichael/buy-my-beer#19]
We look up a User based on their GitHub email address.  If their address at GitHub changes, we can't look them up.

app/models/user.rb

``` ruby
  User.find_by_email(data['email'])
```

Ideas:
- Just print a warning?  "If you have changed your email address on GH ... "
- Use the GH user id instead?  (Could it change too?)

**Comment by franckverrot** [comment]
The user id can't change on GH, this is probably the only unique thing that can't change (the username can change, once only).

**Comment by richardkmichael** [comment]
I'd still need some absolute beer-app UI ability to update this info probably, as a total failsafe .. In case someone on GH creates a new account (GH UID change) and says "this is now me".

But it's a small corner-case.  I was indeed thinking about using the GH UID.

**Comment by franckverrot** [comment]
This is so not supposed to happen you should consider asking people to open a ticket in your very own ticketing system and fix this manually. I would do so.

**Comment by richardkmichael** [comment]
Yes, perhaps a good tradeoff of time/money to implement a solution for a tiny corner-case.  Just fix it by hand as a support ticket untils it's a common enough problem.
